### PR TITLE
[MAINTENANCE] Add test_expectation_suite_send_usage_message

### PR DESCRIPTION
**Pull request by anthonyburdi**
Changes proposed in this pull request:
- Add unit test for `ExpectationSuite.send_usage_message()`
- Showing a mock pattern using a mock/spy to ensure the right events are sent.

### Definition of Done

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.

### CODE REVIEW

**Review by Kilo59 — APPROVED** [review]
🙌

File: tests/core/test_expectation_suite.py [hunk 1]
```diff
@@ -603,3 +604,55 @@ def test_get_expectations_by_domain_type(
...
+        pytest.param(False, id="success=False"),
+    ]
+)
+def test_expectation_suite_send_usage_message(success: bool):
+    """Ensure usage stats event is sent on expectation suite."""
+
+    dc_message_spy = DataContextSendUsageMessageSpy()
+
+    suite = ExpectationSuite(
+        expectation_suite_name="warning",
```
Comment (billdirks) [inline tests/core/test_expectation_suite.py]: nit: maybe change `warning` to `suite` or something more innocuous.

File: tests/core/test_expectation_suite.py [hunk 2]
```diff
@@ -603,3 +604,55 @@ def test_get_expectations_by_domain_type(
...
+        meta={"notes": "This is an expectation suite."},
+        data_context=dc_message_spy,
+    )
+
+    suite.send_usage_event(success=success)
+
+    assert dc_message_spy.messages
+    assert len(dc_message_spy.messages) == 1
+    assert dc_message_spy.messages[0] == {
+        "event": UsageStatsEvents.EXPECTATION_SUITE_ADD_EXPECTATION.value,
```
Comment (billdirks) [inline tests/core/test_expectation_suite.py:651]: This isn't blocking, just a question. Do we want other tests/parameterizations that send in event and event_payload also? 
You could parameterize the inputs as a `parameterized_kwargs` dict and then call `suite.send_usage_event(**parameterized_kwargs)`. When doing the assert you'd need to check for presence/absence of a key in `parameterized_kwargs` to know whether to use the default value or the parameterized value. If that's ugly, we could have a separate test that uses this spy where you send it a nondefault values for these args.

**Review by billdirks — COMMENTED** [review]
This looks great, thank you!

I've added some comments for you to consider but nothing blocking.

**Review by billdirks — APPROVED** [review]

File: tests/core/test_expectation_suite.py (hunk 2 above)
Comment (anthonyburdi) [reply tests/core/test_expectation_suite.py:651]: Good call - when we test `context.send_usage_message()` I think that makes sense. For this method though `suite.send_usage_event(success: bool)` only the `success` parameter is available, the rest are set within the method call.

**Pull request merged**